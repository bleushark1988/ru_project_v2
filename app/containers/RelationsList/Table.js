import styled from 'styled-components';
import Td from './Td';

const Table = styled.table`
  width: 100%;
  background: #fff;
  margin: 1em 0;
  border: 1px solid rgba(34, 36, 38, 0.15);
  box-shadow: none;
  border-radius: .28rem;
  text-align: left;
  color: rgba(0,0,0,.87);
  border-collaps 0, 0, 0eparate;
  border-spacing: 0;
  font-size: 14px;

  tr:first-child ${Td.selector} {
    border-top: none;
  }
`;

export default Table;
