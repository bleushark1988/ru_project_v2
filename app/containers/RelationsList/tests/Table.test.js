import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import 'jest-styled-components';

import Table from '../Table';

describe('<Table />', () => {
  it('should match the snapshot', () => {
    const renderedComponent = renderer.create(<Table />).toJSON();
    expect(renderedComponent).toMatchSnapshot();
  });

  it('should have a className attribute', () => {
    const renderedComponent = shallow(<Table />);
    expect(renderedComponent.prop('className')).toBeDefined();
  });

  it('should adopt a valid attribute', () => {
    const id = 'test';
    const renderedComponent = shallow(<Table id={id} />);
    expect(renderedComponent.prop('id')).toEqual(id);
  });

  it('should not adopt an invalid attribute', () => {
    const renderedComponent = shallow(<Table attribute="test" />);
    expect(renderedComponent.prop('attribute')).toBeUndefined();
  });
});
