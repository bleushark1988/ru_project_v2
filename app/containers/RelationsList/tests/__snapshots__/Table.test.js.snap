// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Table /> should match the snapshot 1`] = `
.c0 {
  width: 100%;
  background: #fff;
  margin: 1em 0;
  border: 1px solid rgba(34,36,38,0.15);
  box-shadow: none;
  border-radius: .28rem;
  text-align: left;
  color: rgba(0,0,0,.87);
  border-collaps: 0,0,0eparate;
  border-spacing: 0;
  font-size: 14px;
}

.c0 tr:first-child {
  border-top: none;
}

<table
  className="c0"
/>
`;
